cnf-certification-test/performance/suite.go

```diff
@@ -72,13 +72,20 @@ func testSchedulingPolicyInCPUPool(env *provider.TestEnvironment,
 		pids := crclient.GetPidsFromPidNamespace(pidNamespace, testContainer)
 
 		// Check for the specified priority for each processes running in that pid namespace
-		if scheduling.ProcessPidsCPUScheduling(pids, testContainer, nonCompliantContainers, schedulingType) {
+		if scheduling.ProcessPidsCPUScheduling(pids, testContainer, nonCompliantContainersMap, schedulingType) {
 			compliantContainers = append(compliantContainers, testContainer)
 		}
 		logrus.Debugf("Processed %v", testContainer)
 	}
-	if len(nonCompliantContainers) != 0 {
+	if len(nonCompliantContainersMap) != 0 {
+		nonCompliantContainers := []*provider.Container{}
+		for k := range nonCompliantContainersMap {
+			nonCompliantContainers = append(nonCompliantContainers, k)
+		}
 		testhelper.AddTestResultLog("Non-compliant", nonCompliantContainers, tnf.ClaimFilePrintf, ginkgo.Fail)
 	}
+	if len(compliantContainers) == 0 {
+		ginkgo.Fail("Failed - no container satisfied scheduling requirements")
+	}
 	tnf.ClaimFilePrintf("Compliant", compliantContainers)
```

Comment:
I would add this log before the AddTestResultLog, otherwise it will never print in case of failure of the test